### PR TITLE
copr: support group by enum column

### PR DESCRIPTION
**Pull request by wshwsh12**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

If I run an aggregate function group by enum type, tikv will panic.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Integration test
- Manual test (add detailed scripts or steps below)

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- No release note.

### CODE REVIEW

File: components/tidb_query_executors/src/fast_hash_aggr_executor.rs [hunk 1]
```diff
@@ -279,7 +280,7 @@ impl<Src: BatchExecutor> AggregationExecutorImpl<Src> for FastHashAggregationImp
                 let group_by_logical_rows = value.logical_rows_struct();
 
                 match_template::match_template! {
-                    TT = [Int, Real, Duration, Decimal, DateTime],
+                    TT = [Int, Real, Duration, Decimal, DateTime, Enum],
```
Comment (Xuanwo) [inline components/tidb_query_executors/src/fast_hash_aggr_executor.rs:283]: Do we need unit tests for hash enum?

**Review by skyzh — APPROVED** [review]

**Review by lzmhhh123 — APPROVED** [review]

File: components/tidb_query_executors/src/fast_hash_aggr_executor.rs (hunk 1 above)
Comment (wshwsh12) [reply components/tidb_query_executors/src/fast_hash_aggr_executor.rs:283]: How should I test it?

**Comment by wshwsh12** [comment]
@Xuanwo  PTAL

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**